Frameworks/CoreGraphics/CGContext.mm

```diff
@@ -2089,6 +2089,12 @@ void CGContextSetPatternPhase(CGContextRef context, CGSize phase) {
             // First glyph's origin is at the given relative position for the glyph run
             CGPoint runningPosition{ glyphRuns[i].relativePosition.x, std::round(glyphRuns[i].relativePosition.y) };
             for (size_t j = 0; j < run->glyphCount; ++j) {
+                if (run->bidiLevel & 1) {
```

Comment:
nit: Since we use this in two different places, but with flipped senses, it might be reasonable to abstract it out into a helper macro or inlinable function. `if(!RUN_IS_BIDI(...))` seems clearer than that we currently have.